lib/Room.php

```diff
@@ -484,6 +484,29 @@ public function setParticipantType($participant, $participantType) {
 		]));
 	}
 
+	/**
+	 * @param Participant $participant
+	 * @param int $participantType
+	 */
+	public function setParticipantTypeBySession(Participant $participant, int $participantType) {
+		$this->dispatcher->dispatch(self::class . '::preSetParticipantTypeBySession', new GenericEvent($this, [
+			'participant' => $participant,
+			'newType' => $participantType,
+		]));
+
+		$query = $this->db->getQueryBuilder();
+		$query->update('talk_participants')
+			->set('participant_type', $query->createNamedParameter($participantType, IQueryBuilder::PARAM_INT))
+			->where($query->expr()->eq('room_id', $query->createNamedParameter($this->getId(), IQueryBuilder::PARAM_INT)))
+			->andWhere($query->expr()->eq('session_id', $query->createNamedParameter($participant->getSessionId())));
+		$query->execute();
+
+		$this->dispatcher->dispatch(self::class . '::postSetParticipantTypeBySession', new GenericEvent($this, [
```

Comment:
Could you please add a listener to notifiy the standalone backend server that the participant has been modified? You can use this as base: https://github.com/nextcloud/spreed/blob/c1d4482d507e44929037b3495cccd63fa5ba0d38/lib/AppInfo/Application.php#L186

Reply:
done